src/utils.rs

```diff
@@ -119,4 +137,59 @@ mod test {
             assert!(x.iter().all(|xi| *xi == 0));
         }
     }
+
+    #[test]
+    fn test_add_le_zero() {
+        for i in 1usize..256 {
+            let mut x = vec!(0u8; i);
+            let mut y = vec!(0u8; i);
+            y[0] = 42;
+            add_le(&mut x, &y);
+            assert!(!x.iter().all(|x| { *x == 0 }));
+            assert_eq!(x, y);
+        }
+    }
+
+    #[test]
+    fn test_add_le_vectors() {
+        let mut x = [255, 2, 3, 4, 5];
+        let y = [42, 0, 0, 0, 0];
+        let z = [41, 3, 3, 4, 5];
+        add_le(&mut x, &y);
+        assert!(!x.iter().all(|x| { *x == 0 }));
+        assert_eq!(x, z);
+        let mut x = [255, 255, 3, 4, 5];
+        let z = [41, 0, 4, 4, 5];
+        add_le(&mut x, &y);
+        assert!(!x.iter().all(|x| { *x == 0 }));
+        assert_eq!(x, z);
+        let mut x = [255, 255, 255, 4, 5];
+        let z = [41, 0, 0, 5, 5];
+        add_le(&mut x, &y);
+        assert!(!x.iter().all(|x| { *x == 0 }));
+        assert_eq!(x, z);
+        let mut x = [255, 255, 255, 255, 5];
+        let z = [41, 0, 0, 0, 6];
+        add_le(&mut x, &y);
+        assert!(!x.iter().all(|x| { *x == 0 }));
+        assert_eq!(x, z);
+        let mut x = [255, 255, 255, 255, 255];
+        let z = [41, 0, 0, 0, 0];
+        add_le(&mut x, &y);
+        assert!(!x.iter().all(|x| { *x == 0 }));
+        assert_eq!(x, z);
+    }
+
+    #[test]
+    fn test_add_le_overflow() {
```

Comment:
What are you testing for here?

Reply:
The same thing that is tested [here](https://github.com/sodiumoxide/sodiumoxide/blob/master/src/utils.rs#L115) - that `add_le` works as modular addition.